pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -187,38 +207,79 @@ func (r *Reconciler) getOrCreateClusterSchedulingPolicySnapshot(ctx context.Cont
 	return latestPolicySnapshot, nil
 }
 
-func (r *Reconciler) deleteRedundantSchedulingPolicySnapshots(ctx context.Context, crp *fleetv1beta1.ClusterResourcePlacement) error {
+func (r *Reconciler) deleteRedundantSchedulingPolicySnapshots(ctx context.Context, crp *fleetv1beta1.ClusterResourcePlacement, revisionHistoryLimit int) error {
 	sortedList, err := r.listSortedClusterSchedulingPolicySnapshots(ctx, crp)
 	if err != nil {
 		return err
 	}
+	if len(sortedList.Items) < revisionHistoryLimit {
+		return nil
+	}
 
-	crpKObj := klog.KObj(crp)
-	// respect the revisionHistoryLimit field
-	revisionLimit := fleetv1beta1.RevisionHistoryLimitDefaultValue
-	if crp.Spec.RevisionHistoryLimit != nil {
-		revisionLimit = *crp.Spec.RevisionHistoryLimit
-		if revisionLimit <= 0 {
-			err := fmt.Errorf("invalid clusterResourcePlacement %s: invalid revisionHistoryLimit %d", crpKObj, revisionLimit)
-			klog.ErrorS(controller.NewExpectedBehaviorError(err), "Invalid revisionHistoryLimit value and using default value instead", "clusterResourcePlacement", crpKObj)
-			// use the default value instead
-			revisionLimit = fleetv1beta1.RevisionHistoryLimitDefaultValue
+	if len(sortedList.Items)-revisionHistoryLimit > 0 {
```

Comment:
Hi Zhiying! The earlier check has deemed that the `revisionHistoryLimit` has been exceeded; this if block is not necessary I think.

Reply:
And we still need to perform cleanup when len(sorted) == revisionHistoryLimit if I understand it correctly.

Reply:
if len(sorted) == limit, that's the expected situation. So we don't need this warning but still need to delete one snapshot.

Added a comment in the code.

Reply:
My concern is, is this really a warning? And the equality is not an expected situation. The limit is just for cleaning thing up.

Reply:
But anyway, this is a minor point.